### ISSUE TITLE
An attempt to dataize the delegate (but not always!)

### ISSUE THREAD

**Issue by dours** [objectionary/eo#621]
This test fails because of an attempt to dataize the `return` object
```
[] > myArrayTest0
  [result] > return
  [] > xmyArray
    return (*.length) > @
  xmyArray.result.eq 0 > @
```

But these further attempts to minimize the test do not fail:
```
[] > myArrayTest0
  [result] > return
  return (*.length) > xmyArray
  xmyArray.result.eq 0 > @
```

```
[] > myArrayTest0
  [result] > return
  (return (*.length)).result.eq 0 > @
```

```
[] > myArrayTest0
  [result] > return
  [] > xmyArray
    return 0 > @
  xmyArray.result.eq 0 > @
```

```
[] > myArrayTest0
  [] > xmyArray
    (*.length) > @
  xmyArray.eq 0 > @
```

**Comment by dours** [comment]
@yegor256 another example for the same problem:
```
[] > myArrayTest0
  [result] > return
  seq > sobaka
    return (*.length)
  sobaka.result.eq 0 > @

```
Here I also cannot remove `seq`, nor change `*.length` to, say, `0`.

**Comment by yegor256** [comment]
@dours this is expected behavior. Consider this code:

```
1 > z

[x] > return

[] > a
  return > @
    z

a.x.eq 1
```

First, when we do `a.x`, we get an object, which is `z`. What's important is that `z.^` points to `a`, not to `return`.

Then, we do `z.eq 1` (remember, the `^` is `a`) and we get `a.eq 1`, which lead to `return.eq 1`.

Dataization of which is an error.

**Comment by yegor256** [comment]
@rultor release, tag is `0.23.3`

**Comment by yegor256** [comment]
@dours fixed, please wait for 0.23.3 release and use it.
